src/main/java/org/chocosolver/solver/constraints/nary/cumulative/literature/vilim2009/PropDisjunctive.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2019, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.cumulative.literature.vilim2009;
+
+import org.chocosolver.solver.constraints.nary.cumulative.literature.CumulativeFilter;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.Task;
+
+import java.util.Comparator;
+import java.util.LinkedList;
+
+
+/**
+ * Disjunctive constraint filtering algorithms described in the thesis :
+ * Vilim, P.: Global constraints in scheduling. Ph.D. thesis, Charles University in Prague (2007). http://vilim.eu/petr/disertace.pdf
+ *
+ * @author Arthur Godet <arth.godet@gmail.com>
+ * @since 23/05/2019
+ */
+public class PropDisjunctive extends CumulativeFilter {
```

Comment:
The class name should not be so generic, there are many other `PropDisjunctive` classes, that may be confusing.